File: factorial.java
import java.util.*;

class factorial{

     int factors(int n){
        if(n == 0){
            return 1;

        }

        int smaller = factors(n-1);
        int bigger = n * smaller;
        return bigger;
        
    }

    public static void main(String []args){

Scanner sc = new Scanner(System.in);
int n = sc.nextInt();

int ans = factors(n);
System.out.println(ans);

    }
}

File: factorial.c
#include <stdio.h>

int factors(int n) {
    if (n == 0) {
        return 1;
    }
    return n * factors(n - 1);
}

int main() {
    int n;
    scanf("%d", &n);
    
    int ans = factors(n);
    printf("%d\n", ans);
    
    return 0;
}
